### PR TITLE
fix(panel): correct footer padding and layout

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #9858 #9857

## Summary

- fixes footer-end slot alignment when its the only thing slotted
- refactor styles
- makes footer slot take precedence over the other footer slots (including deprecated footer-actions)
- add screenshot tests
- cleanup screenshot tests

### CODE REVIEW

**Review by jcfranco — APPROVED** [review]
Code changes LGTM!

Before merging, can you revisit the PR title so that it covers both changes as a single fix or feature? Otherwise, we'll need to split this up.

**Review by macandcheese — APPROVED** [review]
Looks good pending Chromatic approval from @SkyeSeitz @ashetland - Just confirming this gap is not expected to have a css prop for override. I think users can just place a single div and style that as needed for further customization.

**Comment by SkyeSeitz** [comment]
> Looks good pending Chromatic approval from @SkyeSeitz @ashetland - Just confirming this gap is not expected to have a css prop for override. I think users can just place a single div and style that as needed for further customization.

Agreed - I think this makes sense since adding a div to adjust spacing was already the workflow. Now the default is just more accurate. I will remove the css props from the Figma specs.

**Review by SkyeSeitz — APPROVED** [review]

**Pull request merged**
